packages/web/server/api/edge-routers/assets-router.ts

```diff
@@ -105,7 +105,10 @@ export const assetsRouter = createTRPCRouter({
         asset: userAsset,
       });
 
-      return userMarketInfoAsset;
+      return {
+        ...userAsset,
+        ...userMarketInfoAsset,
+      };
```

Comment:
This shold not be needed, as `getAssetMarketInfo` should include the user asset data. Was that not the case?

Reply:
Nope, "amount" field was missing

Reply:
I think amount is missing if the user has no balance for that asset

Reply:
Yes I checked the balance, without the spread it always came back zero, instead with the use of the spread operator between the two objects I get the balance.
Yes it is better to return formatted object with zero value anyway, so we also bring back here the fact that we do not have to hold a reference to fiat currency

Reply:
OK I'll need to fix this

Reply:
@DavideSegullo fixed in stage

Reply:
Perfect, I'll rebase to address It, thank you!